db/migrate/20240312122903_make_stripe_payment_intent_generic.rb

```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+
+class MakeStripePaymentIntentGeneric < ActiveRecord::Migration[7.1]
+  def change
+    change_table :stripe_payment_intents do |t|
+      t.rename :user_id, :initiated_by_id # Make the purpose of the user_id column more clear
+      t.rename :confirmed_by_id, :confirmation_source_id
+      t.rename :confirmed_by_type, :confirmation_source_type
+      t.rename :canceled_by_id, :cancellation_source_id
+      t.rename :canceled_by_type, :cancellation_source_type
+
+      t.string :wca_status
+      t.string :payment_record_type
+      t.integer :payment_record_id
+    end
+
+    rename_table :stripe_payment_intents, :payment_intents
+    rename_column :stripe_records, :status, :stripe_status
+    rename_column :stripe_records, :api_type, :record_type
```

Comment:
Why are you renaming this?

Reply:
Seems more intuitive to me to call it the type of the record as opposed to the type of API it came from (I see the argument for both though, not a hill I'll die on)

Reply:
The reason I'm asking is because this collides with some of our other tables.

In `PaymentIntent` we use `record_type` to define "Is it Stripe or PayPal?" whereas now, inside the Stripe tables, there's _also_ a column called `record_type` which defines something else...

Reply:
That's why I argued to stick with `api_type`, but also any other non-`record-type` name will do :)

Reply:
In line with the `status` -> `stripe_status` change above, we should either:
1. Change `status` to `api_status` XOR
2. Change `api_type` to `stripe_type`

Reply:
Call me crazy, I'm just calling it `type`. It will always be referenced (1) relative to its parent object (`payment_record.type`) or (2) in the scope of the model file (`self.type` or just `type`) so I think this is the least ambiguous way of doing it 

The only reason for `status` -> `stripe_status` is to differentiate it from `wca_status` in similar contexts

Reply:
Going with `stripe_record_type`